src/main/java/com/olegpruh/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,64 @@
 package com.olegpruh.bookstore.service.impl;
 
+import com.olegpruh.bookstore.dto.BookSearchParametersDto;
+import com.olegpruh.bookstore.dto.request.CreateBookRequestDto;
+import com.olegpruh.bookstore.dto.response.BookDto;
+import com.olegpruh.bookstore.mapper.BookMapper;
 import com.olegpruh.bookstore.model.Book;
 import com.olegpruh.bookstore.repository.BookRepository;
+import com.olegpruh.bookstore.repository.BookSpecificationBuilder;
 import com.olegpruh.bookstore.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.stream.Collectors;
+import javax.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final BookSpecificationBuilder bookSpecificationBuilder;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookRepository.save(bookMapper.toModel(requestDto));
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BookDto findBookById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id));
+        return bookMapper.toDto(book);

```

Comment:
use Optional#map to reduce code.
```suggestion
        return bookRepository.findById(id)
        .map(bookMapper::toDto)
        .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id));
```